packages/aws-rfdk/lib/lambdas/nodejs/configure-spot-event-plugin/conversion.ts

```diff
@@ -122,79 +101,35 @@ export function convertTagSpecifications(tagSpecifications: SpotFleetTagSpecific
+  validateStringOptional(launchTemplateOverrides.AvailabilityZone, `${propertyName}.AvailabilityZone`);
+  validateStringOptional(launchTemplateOverrides.InstanceType, `${propertyName}.InstanceType`);
+  validateStringOptional(launchTemplateOverrides.SpotPrice, `${propertyName}.SpotPrice`);
+  validateStringOptional(launchTemplateOverrides.SubnetId, `${propertyName}.SubnetId`);
+  validateProperty(num => num === undefined || typeof num === 'number', launchTemplateOverrides.WeightedCapacity, `${propertyName}.WeightedCapacity`);
 }
 
-export function convertLaunchSpecifications(launchSpecifications: LaunchSpecification[], propertyName: string): LaunchSpecification[] {
-  validateArray(launchSpecifications, propertyName);
+export function validateLaunchTemplateConfigs(launchTemplateConfigs: LaunchTemplateConfig[], propertyName: string): LaunchTemplateConfig[] {
+  validateArray(launchTemplateConfigs, propertyName);
 
-  const convertedLaunchSpecifications: LaunchSpecification[] = [];
-  launchSpecifications.map(launchSpecification => {
-    const SecurityGroups = convertSecurityGroups(launchSpecification.SecurityGroups, `${propertyName}.SecurityGroups`);
-    const TagSpecifications = convertTagSpecifications(launchSpecification.TagSpecifications, `${propertyName}.TagSpecifications`);
-    const BlockDeviceMappings = launchSpecification.BlockDeviceMappings ?
-      convertBlockDeviceMapping(launchSpecification.BlockDeviceMappings, `${propertyName}.BlockDeviceMappings`) : undefined;
+  launchTemplateConfigs.forEach(ltc => {
+    validateProperty(input => input !== undefined && typeof input === 'object' || !Array.isArray(input), ltc.LaunchTemplateSpecification, `${propertyName}.LaunchTemplateSpecification`);
+    validateLaunchTemplateSpecification(ltc.LaunchTemplateSpecification, `${propertyName}.LaunchTemplateSpecification`);
 
-    const convertedLaunchSpecification: LaunchSpecification = {
-      BlockDeviceMappings,
-      IamInstanceProfile: convertInstanceProfile(launchSpecification.IamInstanceProfile, `${propertyName}.IamInstanceProfile`),
-      ImageId: validateString(launchSpecification.ImageId, `${propertyName}.ImageId`),
-      KeyName: validateStringOptional(launchSpecification.KeyName, `${propertyName}.KeyName`),
-      SecurityGroups,
-      SubnetId: validateStringOptional(launchSpecification.SubnetId, `${propertyName}.SubnetId`),
-      TagSpecifications,
-      UserData: validateString(launchSpecification.UserData, `${propertyName}.UserData`),
-      InstanceType: validateString(launchSpecification.InstanceType, `${propertyName}.InstanceType`),
-    };
-    convertedLaunchSpecifications.push(convertedLaunchSpecification);
+    validateProperty(input => Array.isArray(input), ltc.Overrides, `${propertyName}.Overrides`);
+    ltc.Overrides.forEach(override => validateLaunchTemplateOverrides(override, `${propertyName}.Overrides`));
   });
```

Comment:
Can we include the array index in the error message? I think we can get it by accepting an additional argument in the callback for `launchTemplateConfigs.forEach`:

```ts
launchTemplateConfigs.forEach((ltc, index) => {
   // ...
   validateProperty(..., ..., `${propertyName}[${index}].LaunchTemplateSpecification`);
});
```